launchable/commands/subset.py

```diff
@@ -199,18 +205,33 @@ def run(self):
                     res = client.request("post", path, data=gzip.compress(json.dumps(
                         payload).encode()), headers=headers, timeout=timeout)
                     res.raise_for_status()
-
                     output = res.json()["testPaths"]
                 except Exception as e:
                     if os.getenv(REPORT_ERROR_KEY):
                         raise e
                     else:
                         click.echo(e, err=True)
-                    click.echo(click.style("Warning: the service failed to subset. Falling back to running all tests", fg='yellow'), err=True)
-
+                    click.echo(click.style(
+                        "Warning: the service failed to subset. Falling back to running all tests", fg='yellow'), err=True)
 
             # regardless of whether we managed to talk to the service
             # we produce test names
-            click.echo(self.separator.join(self.formatter(t) for t in output))
+            if rest != "":
+                rests = []
+
+                subset = [self.formatter(t) for t in output]
+                for test_path in self.test_paths:
+                    p = self.formatter(test_path)
+                    if p not in subset:
+                        rests.append(p)
+
+                if len(rests) == 0:
+                    click.echo(click.style(
+                        "Warning: remainder files don't exists.", fg='yellow'),  err=True)
```

Comment:
Hmm, it's really interesting what we should do in this case!

Producing an empty file does make sense, but then when I think about, say, Gradle, if this file is empty we'd end up running all the tests, as opposed to none. Maybe we issue a warning and then put one test case in here, as an approximation of "empty test set"?

The point of issuing a message is to drive some actions, but in this case it's not clear what we want the user to do in reaction to this. IOW, what would be an actionable message? If there's no action to be taken, is there any point in telling them?

Either way, the message needs to change. It's not the files that do not exist as they will be created anyway. The warning is that there's no test in the "rest" set. So maybe "Warning: no tests will be in the "rest" file {}"?

I'm inclined to:
- move the last test case from subset to the rest to avoid empty file
- remove the warning
- maybe set something in the subset request so that we can deal with this situation on the server side to understand the extent of this problem, if any.

Reply:
> move the last test case from subset to the rest to avoid empty file
> remove the warning


I agree that.
I felt a little strange before but I’ll be changed back to avoid test fail.